web/src/components/onboarding/OnboardingTaxId.test.tsx

```diff
@@ -139,18 +162,109 @@ describe("<OnboardingTaxId />", () => {
       expect(screen.queryByLabelText("Tax id location")).not.toBeInTheDocument();
     });
 
-    it("clears the full field if user edits the tax id field with * still present", () => {
+    it("renders the hide button initially if the tax id is empty", () => {
       renderComponent(
         {
           ...profileData,
           taxId: "",
         },
         errorProps
       );
-      fireEvent.click(screen.getByLabelText("Tax id"));
-      fireEvent.change(screen.getByLabelText("Tax id"), { target: { value: "******123456" } });
-      fireEvent.blur(screen.getByLabelText("Tax id"));
-      expect((screen.getByLabelText("Tax id") as HTMLInputElement).value).toEqual("");
+      expect(screen.getByText(Config.tax.hideButtonText)).toBeInTheDocument();
+    });
+
+    it("disables the field if the encrypted tax id is populated", () => {
+      renderComponent(
+        {
+          ...profileData,
+          taxId: "********9000",
+          encryptedTaxId: "some-encrypted-value",
+        },
+        errorProps
+      );
+      expect(screen.getByLabelText("Tax id")).toBeDisabled();
+    });
+
+    it("decrypts the tax id if needed when show is clicked and button toggles text to hide", async () => {
```

Comment:
what does the "if needed" here mean? that sounds like it should be two tests describing two cases?